Tests/LibGL/TestRender.cpp

```diff
@@ -1,60 +1,100 @@
 /*
- * Copyright (c) 2021, Leon Albrecht <leon2002.la@gmail.com>.
+ * Copyright (c) 2021, Leon Albrecht <leon2002.la@gmail.com>
+ * Copyright (c) 2022, Jelle Raaijmakers <jelle@gmta.nl>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
-#include <LibTest/TestCase.h>
-
-#include <AK/Debug.h>
-#include <AK/Format.h>
+#include <AK/LexicalPath.h>
+#include <AK/String.h>
+#include <LibCore/FileStream.h>
 #include <LibGL/GL/gl.h>
 #include <LibGL/GLContext.h>
-#include <LibGfx/BMPWriter.h>
 #include <LibGfx/Bitmap.h>
-#include <LibGfx/Font/FontDatabase.h>
-#include <fcntl.h>
-#include <unistd.h>
+#include <LibGfx/QOIWriter.h>
+#include <LibTest/TestCase.h>
 
-#define RENDER_WIDTH 16
-#define RENDER_HEIGHT 16
+#ifdef __serenity__
+#    define REFERENCE_IMAGE_DIR "/usr/Tests/LibGL/reference-images"
+#else
+#    define REFERENCE_IMAGE_DIR "reference-images"
+#endif
+#define SAVE_OUTPUT false
 
-TEST_CASE(simple_triangle)
+static NonnullOwnPtr<GL::GLContext> create_testing_context(int width, int height)
 {
-    auto bitmap = MUST(Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRx8888, { RENDER_WIDTH, RENDER_HEIGHT }));
+    auto bitmap = MUST(Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRx8888, { width, height }));
     auto context = GL::create_context(*bitmap);
-
     GL::make_context_current(context);
+    return context;
+}
+
+static void expect_bitmap_equals_reference(NonnullOwnPtr<GL::GLContext> context, StringView test_name)
```

Comment:
Why don't we simply pass a bitmap as first argument?

Reply:
I disliked having `context->present()` in every test, but needing to `move()` the context into the function doesn't make a lot of sense in hindsight. Changed it so it accepts a `Gfx::Bitmap` instead!